sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -164,9 +168,16 @@ def tune(
         self,
         # TODO (andreyvelich): How to be consistent with other APIs (name) ?
         name: str,
-        objective: Callable,
-        parameters: Dict[str, Any],
-        base_image: str = constants.BASE_IMAGE_TENSORFLOW,
+        model_provider_parameters: Optional[Any] = None,
+		dataset_provider_parameters: Optional[Any] = None,
```

Comment:
Since we only can set the specific types for model and dataset providers should we have one of Union type here ?

Reply:
Yes, I think it will be better. I will adjust that.